### PR TITLE
#452 - user profile cleaning

### PR DESCRIPTION
**Pull request by kamilpiech97**
Connected with #452.
Removed form user profile model/table not used fields.
Updated seeders and tests.
Added cache for user history stuff.

### CODE REVIEW

File: app/Observers/UserHistoryObserver.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+namespace Toby\Observers;
+
+use Laragear\CacheQuery\Facades\CacheQuery;
+use Toby\Models\UserHistory;
+
+class UserHistoryObserver
+{
+    public function creating(UserHistory $userHistory): void
+    {
+        CacheQuery::forget("userHistories{$userHistory->user->id}");
```
Comment (krzysztofrewak) [inline app/Observers/UserHistoryObserver.php:14]:
```suggestion
        CacheQuery::forget("user:history:{$userHistory->user->id}");
```
I think that would be easier to manage.

File: database/migrations/2024_07_04_130731_cleaning_fields_in_profiles_table.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    public function down(): void
+    {
+        Schema::table("profiles", function (Blueprint $table): void {
+            $table->date("last_medical_exam_date")->nullable()->default(null);
+            $table->date("next_medical_exam_date")->nullable()->default(null);
+            $table->date("last_ohs_training_date")->nullable()->default(null);
+            $table->date("next_ohs_training_date")->nullable()->default(null);
+            $table->date("employment_date")->nullable();
```
Comment (krzysztofrewak) [inline database/migrations/2024_07_04_130731_cleaning_fields_in_profiles_table.php:28]: No default value?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: app/Observers/UserHistoryObserver.php (hunk 1 above)
Comment (kamilpiech97) [reply app/Observers/UserHistoryObserver.php:14]: Should I update other cache keys?

File: database/migrations/2024_07_04_130731_cleaning_fields_in_profiles_table.php (hunk 2 above)
Comment (kamilpiech97) [reply database/migrations/2024_07_04_130731_cleaning_fields_in_profiles_table.php]: added

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Review by krzysztofrewak — APPROVED** [review]

**Pull request merged**